### PR TITLE
improving the cpu/mem section

### PR DESCRIPTION
**Pull request by choldgraf**
@yuvipanda mentioned that this section could use another pair of eyes to clean it up. There are a few spots that need input about specific syntax, LMK what you think.

### CODE REVIEW

File: doc/source/extending-jupyterhub.rst [hunk 1]
```diff
@@ -44,31 +45,60 @@ To instruct JupyterHub to use this image, simply add this to your ``config.yaml`
...
-Each user on your JupyterHub gets a slice of memory and CPU to use. By default, each user is
-*guaranteed* 1G of RAM, but no enforced limits. This means each user is guaranteed at minimum
-1G of RAM, but no exact upper limit (this will vary based on various factors outside of your
-control). We recommend most people change this!
+Each user on your JupyterHub gets a slice of memory and CPU to use. There are
+two ways in which you can provision these resources: as *guaranteed* resources,
+or as *limited* resources.
+
+A *guaranteed* resource means that all users will have *at least* this resource
+available at all times, but they may be able to use more resource if they're
```
Comment (willingc) [inline doc/source/extending-jupyterhub.rst]: ~~able to use~~ given ... resource/s

File: doc/source/extending-jupyterhub.rst [hunk 2]
```diff
@@ -44,31 +45,60 @@ To instruct JupyterHub to use this image, simply add this to your ``config.yaml`
...
+available. For example, if users are *guaranteed* 1G of RAM, users can technically
+use more than 1G of RAM if these resources aren't being used by other computers.
 
-If you want all your users to have guaranteed access to 1Gi of RAM, and no more, you can add the
-following to your config.yaml and do an upgrade:
+A *limited* resource sets a hard limit on the resources available. In the example
```
Comment (willingc) [inline doc/source/extending-jupyterhub.rst]: ~~In the example above,~~ If a memory is set up as a *limited* resource with a 1G memory limit, an individual user could use nor more than 1G of memory (RAM), ...

File: doc/source/extending-jupyterhub.rst [hunk 3]
```diff
@@ -44,31 +45,60 @@ To instruct JupyterHub to use this image, simply add this to your ``config.yaml`
...
+above, if there were a 1G memory limit, it would mean that users could use
+no more than 1G of RAM, no matter what other resources are being used on the 
+machines.
+
+By default, each user is *guaranteed* 1G of RAM. AKA all users have *at least*
```
Comment (willingc) [inline doc/source/extending-jupyterhub.rst]: ~~AKA~~

File: doc/source/extending-jupyterhub.rst [hunk 4]
```diff
@@ -44,31 +45,60 @@ To instruct JupyterHub to use this image, simply add this to your ``config.yaml`
...
+1G, but they can technically use more if it is available. You can easily change
+the amount of these resources, and whether they are guaranteed or limited, by
```
Comment (willingc) [inline doc/source/extending-jupyterhub.rst]: *guaranteed* or *limited*

File: doc/source/extending-jupyterhub.rst [hunk 5]
```diff
@@ -44,31 +45,60 @@ To instruct JupyterHub to use this image, simply add this to your ``config.yaml`
...
+the guarantee if you want to allow some users to use larger amounts of RAM for
+a very short-term time (e.g. when running a single, short-lived function that
+consumes a lot of memory).
 
-Kubernetes will make sure that each user will always have access to 1G of RAM, and requests for
-more RAM will fail (your kernel will usually die). The guarantee is the base amount of RAM that
-is guaranteed for each user, and the limit is the amount of RAM at which new memory requests
-will not be granted. You can set the limit to be higher than the guarantee if you want to allow
-some users to 'burst' RAM use occasionally (and use more RAM than the guarantee).
+If you also wanted to set a guarantee of CPUs, you would do so with the
```
Comment (willingc) [inline doc/source/extending-jupyterhub.rst]: ~~If you also wanted~~ To

**Comment by yuvipanda** [comment]
I've updated the docs changes @willingc suggested, fixed some inaccuracies and am merging it now.

@willingc I think you should feel free to just make these changes yourself without having to wait for chris or me (if you wanted to, of course! These PRs are all marked as editable). What do you think, @choldgraf?

**Comment by choldgraf** [comment]
Thanks @yuvipanda ! This one slipped through the cracks for me since I've been busy w/ graduation stuff. I'm definitely OK with making edits on stale PRs if it means more merging :)

**Pull request merged**